environments.json

```diff
@@ -0,0 +1,123 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Parameters": {
+    "Environment": {
+      "Type": "AWS::SSM::Parameter::Value<String>"
+    }
+  },
+  "Resources": {
+    "DiagramTerraformRole": {
+      "Type": "AWS::IAM::Role",
+      "Properties": {
+        "RoleName": "DiagramTerraformRole",
+        "AssumeRolePolicyDocument": {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Principal": {
+                "Federated": {
+                  "Fn::Sub": "arn:aws:iam::${AWS::AccountId}:oidc-provider/token.actions.githubusercontent.com"
+                }
+              },
+              "Action": "sts:AssumeRoleWithWebIdentity",
+              "Condition": {
+                "StringLike": {
+                  "token.actions.githubusercontent.com:aud": "sts.amazonaws.com",
+                  "token.actions.githubusercontent.com:sub": [
+                    "repo:nationalarchives/DiAGRAM-terraform:ref:refs/heads/*",
+                    {"Fn::Sub" : "repo:nationalarchives/DiAGRAM-terraform:environment:${Environment}"}
+                  ]
+                }
+              }
+            },
+            {
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {"Fn::Sub":  "arn:aws:iam::${AWS::AccountId}:root"}
+              },
+              "Action": "sts:AssumeRole"
+            }
+          ]
+        },
+        "Policies": [
+          {
+            "PolicyName": "DiagramTerraformPolicy",
+            "PolicyDocument": {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Sid": "",
+                  "Effect": "Allow",
+                  "Action": [
+                    "route53:*",
+                    "acm:*",
+                    "cloudfront:*",
+                    "apigateway:*",
+                    "logs:*",
+                    "lambda:*",
+                    "wafv2:*"
+                  ],
+                  "Resource": "*"
```

Comment:
Is it OK that we're this loose with the Resources?

Reply:
Yep